### PR TITLE
 Infoblox scenario guide

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

Closes issue https://github.com/ansible/ansible/issues/42964

Create Infoblox Scenario Guide

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs.ansible.com
##### ANSIBLE VERSION

```paste below
2.8
```

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: docs/docsite/rst/scenario_guides/guide_infoblox.rst [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+Module list
+============
+
+See :ref:`nios_net tools_modules` for a full list of NIOS modules for this release.
+Each module includes simple documented example tasks for how to use them. `Use cases with modules`_ expands on some of these example tasks.
+
+NIOS lookup plugins
+===================
+
+Ansible includes the following lookup plugins for NIOS:
```
Comment (acozine) [inline docs/docsite/rst/scenario_guides/guide_infoblox.rst:44]: Link to https://docs.ansible.com/ansible/latest/plugins/lookup.html might be useful here.

File: docs/docsite/rst/scenario_guides/guide_infoblox.rst [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+    ---
+    nios_provider:
+      host: 192.0.0.2
+      username: admin
+      password: ansible
+
+Module list
+============
+
+See :ref:`nios_net tools_modules` for a full list of NIOS modules for this release.
```
Comment (acozine) [inline docs/docsite/rst/scenario_guides/guide_infoblox.rst]: In the published doc, this is rendered as "See Nios for a full list of NIOS modules for this release." That looks a bit weird to my eye. How about tightening this up and specifying the text for the link - something like:
```
You can review simple example tasks in the documentation for any of the :ref:`NIOS modules <nios_net tools_modules>` or look at the `Use cases with modules` section for more elaborate examples.
```

Comment (acozine) [reply docs/docsite/rst/scenario_guides/guide_infoblox.rst]: Actually, I might get rid of the "Module list" heading and put the shortened content in the Introduction.

File: docs/docsite/rst/scenario_guides/guide_infoblox.rst [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+- :ref:`nios_next_ip <nios_next_ip_lookup>` Provides the next available IP address from a network. You'll see an example of this in `Creating a host record`_.
+- :ref:`nios_next_network <nios_next_network_lookup>` - Returns the next available network range for a network-container.
+
+.. note:: You must run these lookup plugins locally by specifying ``connection: local``.
+
+
+Retrieving all network views
+----------------------------
+
+This example playbook uses the ``set_fact`` module with the ``nios`` lookup plugin to retrieve all the network views, which are then saved in the ``networkviews`` variable:
```
Comment (acozine) [inline docs/docsite/rst/scenario_guides/guide_infoblox.rst]: I think "this example playbook" is fluff - how about rewording these examples using the pattern: To do X, use Y:  <example_playbook>? So for this one:
```
To retrieve all network views and save them in a variable, use the `set_fact` module with the `nios` lookup plugin:
```

File: docs/docsite/rst/scenario_guides/guide_infoblox.rst [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+See the :ref:`nios <nios_lookup>` examples for more data options that you can retrieve.
+
+You can access these playbooks at `Infoblox lookup playbooks <https://github.com/network-automation/infoblox_ansible/tree/master/lookup_playbooks>`_.
+
+Use cases with modules
+======================
+
+You can use ``nios`` modules in tasks to simplify common Infoblox workflows.
+
+For these examples, you need to set up your NIOS credentials. See `Credentials and authenticating`_.
```
Comment (acozine) [inline docs/docsite/rst/scenario_guides/guide_infoblox.rst]: What do you think about shortening this section by putting the link in the sentence?
```
You can use the ``nios`` modules in tasks to simplify common Infoblox workflows. Be sure to set up your `NIOS credentials <nios_credentials>` before following these examples.
```
or something like that?

File: docs/docsite/rst/scenario_guides/guide_infoblox.rst [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+You can use the Infoblox dynamic inventory script to import your network node inventory with Infoblox NIOS. To gather the inventory from Infoblox, you need two files:
+
+- `infoblox.yaml <https://raw.githubusercontent.com/ansible/ansible/devel/contrib/inventory/infoblox.yaml>`_ - A file that specifies the NIOS provider arguments and optional filters.
+
+- `infoblox.py <https://raw.githubusercontent.com/ansible/ansible/devel/contrib/inventory/infoblox.py>`_ - The python script that retrieves the NIOS inventory.
+
+To use the Infoblox dynamic inventory script:
+
+1. Download the ``infoblox.yaml`` file and save it in the ``/etc/ansible`` directory.
```
Comment (acozine) [inline docs/docsite/rst/scenario_guides/guide_infoblox.rst]: Do you want these to be hard-coded numbers, or a #-style list?

File: docs/docsite/rst/scenario_guides/guide_infoblox.rst [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+You can explicitly use the Infoblox dynamic inventory script as follows:
+
+.. code-block:: bash
+
+    $ ansible -i infoblox.py all -m ping
+
+You can also implicitly use the Infoblox dynamic inventory script by including it in your inventory directory (``etc/ansible/hosts`` by default).
+
+See :ref:`dynamic_inventory` for more details.
```
Comment (acozine) [inline docs/docsite/rst/scenario_guides/guide_infoblox.rst]: I feel like the whitespace in the rendered HTML makes it hard to parse the page - maybe try having fewer paragraphs/newlines?

**Review by acozine — COMMENTED** [review]
Nice work. I made a few suggestions for making it even better.

**Comment by abenokraitis** [comment]
Please take a look at the draft in stage:

http://docs.testing.ansible.com/ansible/latest/scenario_guides/guide_infoblox.html

NOTE: This link may or may not work due to QA workflows

File: docs/docsite/rst/scenario_guides/guide_infoblox.rst [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+    localhost                  : ok=5    changed=0    unreachable=0    failed=0
+
+The output above shows the host record for ``leaf01.ansible.com`` and ``leaf02.ansible.com`` that were retrieved by the ``nios`` lookup plugin. This playbook saves the information in variables which you can use in other playbooks. This allows you to use Infoblox as a single source of truth to gather and use information that changes dynamically. See :ref:`playbooks_variables` for more information on using Ansible variables. See the :ref:`nios <nios_lookup>` examples for more data options that you can retrieve.
+
+You can access these playbooks at `Infoblox lookup playbooks <https://github.com/network-automation/infoblox_ansible/tree/master/lookup_playbooks>`_.
+
+Use cases with modules
+======================
+
+You can use the ``nios`` modules in tasks to simplify common Infoblox workflows. Be sure to set up your `NIOS credentials <nios_credentials>`_ before following these examples.
```
Comment (gundalow) [inline docs/docsite/rst/scenario_guides/guide_infoblox.rst]: Currently this looks like a link to a new file, rather than a relative link
http://docs.testing.ansible.com/ansible/latest/scenario_guides/nios_credentials

**Comment by housysadm** [comment]
I really appreciate this document being created.

I was a little confused though by the mention of the WAPI only in one specific spot. I assume the modules can ONLY talk to the WAPI so maybe a statement specifying that near the top and then no other mention of it. 

Would it be appropriate to mention in the prerequisites or authentication section that you need a NIOS account with the WAPI feature enabled? Or is that outside the scope of the document?
